packages/rich-text/src/component/index.native.js

```diff
@@ -401,33 +412,66 @@ export class RichText extends Component {
 		this.lastAztecEventType = 'input';
 	}
 
-	handleMention( event ) {
+	handleTriggerKeyCodes( event ) {
 		const { keyCode } = event;
+		const triggeredOption = this.suggestionOptions().find( ( option ) => {
+			const triggeredKeyCode = option.triggerChar.charCodeAt( 0 );
+			return triggeredKeyCode === keyCode;
+		} );
 
-		if ( keyCode !== '@'.charCodeAt( 0 ) ) {
-			return;
-		}
-		const record = this.getRecord();
-		const text = getTextContent( record );
-		// Only start the mention UI if the selection is on the start of text or the character before is a space
-		if (
-			text.length === 0 ||
-			record.start === 0 ||
-			text.charAt( record.start - 1 ) === ' '
-		) {
-			this.showMention();
-		} else {
-			this.insertString( record, '@' );
+		if ( triggeredOption ) {
+			const record = this.getRecord();
+			const text = getTextContent( record );
+			// Only respond to the trigger if the selection is on the start of text or line
+			// or if the character before is a space
+			const useTrigger =
+				text.length === 0 ||
+				record.start === 0 ||
+				text.charAt( record.start - 1 ) === '\n' ||
+				text.charAt( record.start - 1 ) === ' ';
+
+			if ( useTrigger && triggeredOption.onClick ) {
+				triggeredOption.onClick();
+			} else {
+				this.insertString( record, triggeredOption.triggerChar );
+			}
 		}
 	}
 
-	showMention() {
-		const record = this.getRecord();
-		addMention()
-			.then( ( mentionUserId ) => {
-				this.insertString( record, `@${ mentionUserId } ` );
-			} )
-			.catch( () => {} );
+	suggestionOptions() {
+		const { areMentionsSupported, areXPostsSupported } = this.props;
+		const allOptions = [
+			{
+				supported: areMentionsSupported,
+				title: __( 'Insert mention' ),
+				onClick: this.handleUserSuggestion,
+				triggerChar: '@',
+				value: 'mention',
+				label: __( 'Mention' ),
+				icon: atSymbol,
+			},
+			{
+				supported: areXPostsSupported,
+				title: __( 'Insert crosspost' ),
+				onClick: this.handleXpostSuggestion,
+				triggerChar: '+',
+				value: 'crosspost',
+				label: __( 'Crosspost' ),
+				icon: plus,
+			},
+		];
+		return allOptions.filter( ( op ) => op.supported );
```

Comment:
I personally prefer using _functional_ flows like using `filter` here but I think it might be more familiar to build `allOptions` with  conditionals :
```
const allOptions = [];
if ( areMentionsSupported ) {
  allOptions.push({<option object>})
}

if ( areXPostsSupported ) {
  allOptions.push({<option object>})
}

return allOptions;
``